### PR TITLE
feat: serialize/deserialize support for PromQL plans

### PR DESCRIPTION
**Pull request by waynexia**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Support to send PromQL logical extension plans via substrait. Implement extension serializer registry.

## Checklist

- [x]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

**Comment by waynexia** [comment]
Depends on https://github.com/GreptimeTeam/greptimedb/pull/1660

File: src/promql/src/extension_plan/instant_manipulate.rs [hunk 1]
```diff
@@ -127,6 +135,40 @@ impl InstantManipulate {
...
+            pb_instant_manipulate.end,
+            pb_instant_manipulate.lookback_delta,
+            pb_instant_manipulate.interval,
+            pb_instant_manipulate.time_index,
+            if pb_instant_manipulate.field_index.is_empty() {
+                None
+            } else {
+                Some(pb_instant_manipulate.field_index)
+            },
+            placeholder_plan,
```
Comment (fengjiachun) [inline src/promql/src/extension_plan/instant_manipulate.rs:169]: I prefer to use struct assignment, the `new` function has many parameters and it is easy to get the order wrong.

File: src/query/src/lib.rs [hunk 2]
```diff
@@ -18,6 +18,7 @@ pub mod datafusion;
 pub mod dist_plan;
 pub mod error;
 pub mod executor;
+mod extension_serializer;
```
Comment (etolbakov) [inline src/query/src/lib.rs:21]: should it be `pub mod` here as well?

Comment (waynexia) [reply src/query/src/lib.rs:21]: nice catch!

**Comment by fengjiachun** [comment]
Some conflicts. @waynexia

**Review by killme2008 — APPROVED** [review]
LGTM

**Review by fengjiachun — APPROVED** [review]

**Pull request merged**